server/data/data-library/src/main/java/org/apache/james/repository/file/AbstractFileRepository.java

```diff
@@ -261,19 +261,19 @@ protected OutputStream getOutputStream(final String key) throws IOException {
      * 
      * @param key
      *            the key to remove
+     * @throws IOException 
      */
-    public synchronized void remove(final String key) {
+    
+    public synchronized boolean remove(final String key) {
         try {
             FileUtils.forceDelete(getFile(key));
-            if (DEBUG)
-                getLogger().debug("removed key " + key);
-        } catch (FileNotFoundException e) {
-            getLogger().debug("File for " + key + " not found: wasn't able to remove");
+            return true;
```

Comment:
it's a good start but now you should use this to throw the right exception in the UserRepository case